generator/internal/sample_generator.cc

```diff
@@ -87,6 +89,95 @@ void SetClientEndpoint(std::vector<std::string> const& argv) {
+ public:
+  ~CustomIdempotencyPolicy() override = default;
+  std::unique_ptr<google::cloud::$product_namespace$::$idempotency_class_name$> clone() const override {
+    return std::make_unique<CustomIdempotencyPolicy>(*this);
+  }
+  // Override inherited functions to define as needed.
+};
+//! [custom-idempotency-policy]
+
+void SetRetryPolicy(std::vector<std::string> const& argv) {
+  if (!argv.empty()) {
+    throw google::cloud::testing_util::Usage{"set-client-retry-policy"};
+  }
+  //! [set-retry-policy]
+  auto options = google::cloud::Options{}
+    .set<google::cloud::$product_namespace$::$idempotency_class_name$Option>(
+      CustomIdempotencyPolicy().clone())
+    .set<google::cloud::$product_namespace$::$retry_policy_name$Option>(
+      google::cloud::$product_namespace$::$limited_error_count_retry_policy_name$(3).clone())
+    .set<google::cloud::$product_namespace$::$service_name$BackoffPolicyOption>(
+      google::cloud::ExponentialBackoffPolicy(
+          /*initial_delay=*/std::chrono::milliseconds(200),
+          /*maximum_delay=*/std::chrono::seconds(45),
+          /*scaling=*/2.0).clone());)""");
+  if (HasGenerateGrpcTransport()) {
+    HeaderPrint(R"""(
+  auto connection = google::cloud::$product_namespace$::Make$connection_class_name$()""");
+  } else {
+    HeaderPrint(R"""(
+  auto connection = google::cloud::$product_namespace$::Make$connection_class_name$Rest()""");
+  }
+  if (IsExperimental()) HeaderPrint("google::cloud::ExperimentalTag{}, ");
+  switch (endpoint_location_style) {
+    case ServiceConfiguration::LOCATION_DEPENDENT:
+      HeaderPrint(R"""("location-unused-in-this-example", )""");
+      break;
+    case ServiceConfiguration::LOCATION_DEPENDENT_COMPAT:
+    default:
+      break;
+  }
+  HeaderPrint(R"""(options);)""");
+  if (IsExperimental()) {
+    HeaderPrint(R"""(
+
+  // c1 and c2 share the same retry policies
+  auto c1 = google::cloud::$product_namespace$::$client_class_name$(
+    google::cloud::ExperimentalTag{}, connection);
+  auto c2 = google::cloud::$product_namespace$::$client_class_name$(
+    google::cloud::ExperimentalTag{}, connection);
+
+  // You can override any of the policies in a new client. This new client
+  // will share the policies from c1 (or c2) *except* from the retry policy.
+  auto c3 = google::cloud::$product_namespace$::$client_class_name$(
+    google::cloud::ExperimentalTag{}, connection,
+    google::cloud::Options{}.set<google::cloud::$product_namespace$::$retry_policy_name$Option>(
+      google::cloud::$product_namespace$::$limited_time_retry_policy_name$(std::chrono::minutes(5)).clone()));
+
+  // You can also override the policies in a single call:
+  // c3.SomeRpc(..., google::cloud::Options{}
+  //     .set<google::cloud::$product_namespace$::$retry_policy_name$Option>(
+  //       google::cloud::$product_namespace$::$limited_error_count_retry_policy_name$(10).clone()));
+  //! [set-client-client-retry-policy]
```

Comment:
closing tag does not match starting tag (`set-retry-policy`)

https://github.com/googleapis/google-cloud-cpp/blob/d542a62ea07077f15edb07c6b24f035e7fb038ba/generator/internal/sample_generator.cc#L108

Reply:
Doh! Fixed.